optimum/quanto/models/hub_utils.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Adapted from
+https://github.com/huggingface/diffusers/blob/69e72b1dd113927ed638f26e82738e9735385edc/src/diffusers/utils/hub_utils.py#L503C1-L601C14
+"""
+
+import os
+import tempfile
+from typing import Optional, Union
+
+from huggingface_hub import create_repo, upload_folder
+
+
+class PushToHubMixin:
+    """
+    A Mixin to push a model, scheduler, or pipeline to the Hugging Face Hub.
+    """
+
+    def _upload_folder(
+        self,
+        working_dir: Union[str, os.PathLike],
+        repo_id: str,
+        token: Optional[str] = None,
+        commit_message: Optional[str] = None,
+        create_pr: bool = False,
+    ):
+        """
+        Uploads all files in `working_dir` to `repo_id`.
+        """
+        if commit_message is None:
+            commit_message = f"Upload {self.__class__.__name__}"
+
+        return upload_folder(
+            repo_id=repo_id, folder_path=working_dir, token=token, commit_message=commit_message, create_pr=create_pr
+        )
```

Comment:
I'd not define this `_upload_folder` alias method. It's used only once and it makes the class more complex for almost no benefit. Better to do a `upload_folder` directly in `push_to_hub` and set the correct commit message there.

Reply:
Done in [1cc107f](https://github.com/huggingface/optimum-quanto/pull/263/commits/1cc107f1f12c87ecb3a46ed4f3e307e0e3a291f1).